server/tests/trim_gap_handling.rs

```diff
@@ -329,26 +329,34 @@ async fn applied_lsn_converged(
     Ok(())
 }
 
-async fn grpc_connect(address: AdvertisedAddress) -> Result<Channel, tonic::transport::Error> {
-    match address {
-        AdvertisedAddress::Uds(uds_path) => {
-            // dummy endpoint required to specify an uds connector, it is not used anywhere
-            Endpoint::try_from("http://127.0.0.1")
-                .expect("/ should be a valid Uri")
-                .connect_with_connector(service_fn(move |_: Uri| {
-                    let uds_path = uds_path.clone();
-                    async move {
-                        Ok::<_, io::Error>(TokioIo::new(UnixStream::connect(uds_path).await?))
-                    }
-                })).await
-        }
-        AdvertisedAddress::Http(uri) => {
-            Channel::builder(uri)
-                .connect_timeout(Duration::from_secs(2))
-                .timeout(Duration::from_secs(2))
-                .http2_adaptive_window(true)
-                .connect()
-                .await
+struct TestNetworkOptions {
+    connect_timeout: u64,
+    request_timeout: u64,
+}
+
+impl Default for TestNetworkOptions {
+    fn default() -> Self {
+        Self {
+            connect_timeout: 1000,
+            request_timeout: 5000,
         }
     }
 }
+
+impl CommonClientConnectionOptions for TestNetworkOptions {
+    fn connect_timeout(&self) -> Duration {
+        Duration::from_millis(self.connect_timeout)
+    }
+
+    fn keep_alive_interval(&self) -> Duration {
+        Duration::from_secs(60)
+    }
+
+    fn keep_alive_timeout(&self) -> Duration {
+        Duration::from_millis(self.request_timeout)
+    }
```

Comment:
Same question here. Before we seem to have set `Channel::builder().timeout(request_timeout)` which is no longer the case with this change.

Reply:
Fixed now; I was being blind - I've added a dedicated `request_timeout` property. I messed up when merging the logic of the cli/test-only and production code endpoint building.